app/celery/letters_pdf_tasks.py

```diff
@@ -117,26 +121,31 @@ def get_letters_pdf(template, contact_block, filename, values):
 
 @notify_celery.task(name='collate-letter-pdfs-for-day')
 @cronitor("collate-letter-pdfs-for-day")
-def collate_letter_pdfs_for_day(date=None):
-    if not date:
-        # Using the truncated date is ok because UTC to BST does not make a difference to the date,
-        # since it is triggered mid afternoon.
-        date = datetime.utcnow().strftime("%Y-%m-%d")
-
-    letter_pdfs = sorted(
-        s3.get_s3_bucket_objects(
-            current_app.config['LETTERS_PDF_BUCKET_NAME'],
-            subfolder=date
-        ),
-        key=lambda letter: letter['Key']
-    )
-    for i, letters in enumerate(group_letters(letter_pdfs)):
+def collate_letter_pdfs_for_day():
+    """
+    Finds all letters which are still waiting to be sent to DVLA for printing
+
+    This would usually be run at 5.50pm and collect up letters created between before 5:30pm today
+    that have not yet been sent.
+    If run after midnight, it will collect up letters created before 5:30pm the day before.
+    """
+    date = convert_utc_to_bst(datetime.utcnow())
+    if date.time() < LETTER_PROCESSING_DEADLINE:
+        date = date - timedelta(days=1)
+
+    # Using the truncated date is ok because UTC to BST does not make a difference to the date,
+    # since it is triggered mid afternoon.
+    print_run_date = date.strftime("%Y-%m-%d")
+
+    letters_to_print = get_key_and_size_of_letters_to_be_sent_to_print(print_run_date)
```

Comment:
Wondering if it's best to past a "%Y-%m-%d" string into this function or instead maybe a datetime

Reply:
Makes sense, since the first thing the dao method does is convert it to datetime